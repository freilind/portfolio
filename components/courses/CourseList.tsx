import React, { FC } from "react";
import { Grid, Image, Link } from "@nextui-org/react";
import { ICourse } from "../../interfaces/course";

const CourseList: FC = () => {
  return (
    <>
      {courses.map((course) => (
        <Grid key={course.name} sm={5}>
          <Link href={course.url} target={"_blank"}>
            <Image
              objectFit="fill"
              autoResize={true}
              width={310}
              height={220}
              src={course.pathImage}
              alt={course.name}
            />
          </Link>
        </Grid>
      ))}
    </>
  );
};

export default CourseList;

const courses: Array<ICourse> = [
  {
    name: "certificate-java8",
    pathImage: "/images/courses/cert-java8.jpg",
    url: "/images/courses/cert-java8.jpg",
  },
  {
    name: "next-frmwk-prod",
    pathImage: "/images/courses/next-frmwk-prod.jpeg",
    url: "https://udemy-certificate.s3.amazonaws.com/pdf/UC-625fca48-56d0-499a-abd0-2e3299aeb3a0.pdf",
  },
  {
    name: "next-react-guide",
    pathImage: "/images/courses/next-react-guide.jpeg",
    url: "https://udemy-certificate.s3.amazonaws.com/pdf/UC-ea89ede5-8ad5-4182-bc23-d467a82f7ab4.pdf",
  },
  {
    name: "web-bootcamp2022",
    pathImage: "/images/courses/web-bootcamp2022.jpeg",
    url: "https://udemy-certificate.s3.amazonaws.com/pdf/UC-fa37664f-a127-40f3-b590-64893ff75dee.pdf",
  },
  {
    name: "master-cqrs",
    pathImage: "/images/courses/master-cqrs.jpeg",
    url: "https://udemy-certificate.s3.amazonaws.com/pdf/UC-2486bc7f-f452-4c96-8f0d-9c9e229a7223.pdf",
  },
  {
    name: "flutter-complete-guide",
    pathImage: "/images/courses/flutter-complete-guide.jpeg",
    url: "https://udemy-certificate.s3.amazonaws.com/pdf/UC-5b81bda1-1845-4530-92df-681e05705f46.pdf",
  },
  {
    name: "flutter-2021",
    pathImage: "/images/courses/flutter-2021.jpeg",
    url: "https://udemy-certificate.s3.amazonaws.com/pdf/UC-07aabcbd-8c1e-461d-9518-e69cbf0d998c.pdf",
  },
  {
    name: "docker-guide",
    pathImage: "/images/courses/docker-guide.jpeg",
    url: "https://udemy-certificate.s3.amazonaws.com/pdf/UC-d4ef5915-4608-46ba-bfd4-f7baf95d2346.pdf",
  },
  {
    name: "kafka",
    pathImage: "/images/courses/kafka.jpeg",
    url: "https://udemy-certificate.s3.amazonaws.com/pdf/UC-bca7b665-586d-4d9c-8105-d0546e5a3dad.pdf",
  },
];
