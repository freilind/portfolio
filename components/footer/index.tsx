import React, { FC } from 'react';

const Footer: FC = () => {
    return (
        <footer>
            <p>&copy; {new Date().getFullYear()} All Rights Reserved. </p>
        </footer>
    )
}

export default Footer;
