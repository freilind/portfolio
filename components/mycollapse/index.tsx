import React, { FC } from "react";
import { Collapse, Grid } from "@nextui-org/react";
import { PortfolioList } from "../portfolio";
import { ExperienceList } from "../experience";
import Education from "../education/Education";
import { useIntl } from "react-intl";
import { CourseList } from "../courses";

const MyCollapse: FC = () => {
  const intl = useIntl();

  const title1 = intl.formatMessage({ id: "collapse.title.1" });
  const subtitle1 = intl.formatMessage({ id: "collapse.subtitle.1" });
  const title2 = intl.formatMessage({ id: "collapse.title.2" });
  const subtitle2 = intl.formatMessage({ id: "collapse.subtitle.2" });
  const title3 = intl.formatMessage({ id: "collapse.title.3" });
  const subtitle3 = intl.formatMessage({ id: "collapse.subtitle.3" });
  const title4 = intl.formatMessage({ id: "collapse.title.4" });
  const subtitle4 = intl.formatMessage({ id: "collapse.subtitle.4" });

  return (
    <Grid.Container gap={2}>
      <Grid>
        <Collapse.Group key={"collapse-group"}>
          <Collapse
            key={"collapse1"}
            title={title1}
            subtitle={subtitle1}
            aria-labelledby="collapse1"
          >
            <Grid.Container
              key={"collapse-container-1"}
              gap={4}
              justify="center"
            >
              <PortfolioList />
            </Grid.Container>
          </Collapse>
          <Collapse
            key={"collapse2"}
            title={title2}
            subtitle={subtitle2}
            aria-labelledby="collapse2"
          >
            <Grid.Container
              key={"collapse-container-2"}
              gap={2}
              justify="center"
            >
              <ExperienceList />
            </Grid.Container>
          </Collapse>
          <Collapse
            key={"collapse4"}
            title={title4}
            subtitle={subtitle4}
            aria-labelledby="collapse3"
          >
            <Grid.Container
              key={"collapse-container-4"}
              gap={4}
              justify="center"
            >
              <CourseList />
            </Grid.Container>
          </Collapse>
          <Grid
            xs={0}
            sm={12}
            css={{
              padding: "$0",
            }}
          >
            <Collapse
              key={"collapse3"}
              title={title3}
              subtitle={subtitle3}
              aria-labelledby="collapse4"
            >
              <Grid.Container
                key={"collapse-container-3"}
                gap={1}
                justify="center"
                xs={12}
              >
                <Education />
              </Grid.Container>
            </Collapse>
          </Grid>
        </Collapse.Group>
      </Grid>
    </Grid.Container>
  );
};

export default MyCollapse;
