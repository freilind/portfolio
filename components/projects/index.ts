import { IProject } from "../../interfaces";

export const projectsList: Array<IProject> = [
    {
        url: '/portfolio/github-user-search',
        title: 'Github user search',
        imgLight: '/images/github-search-light.png',
        imgDark: '/images/github-search-dark.png',
    },
    {
        url: '/portfolio/to-do',
        title: 'Todo list',
        imgLight: '/images/todo-light.png',
        imgDark: '/images/todo-dark.png'
    },
    {
        url: '/portfolio/markdown',
        title: 'Markdown',
        imgLight: '/images/markdown-preview-light.png',
        imgDark: '/images/markdown-preview-dark.png'
    },
    {
        url: '/portfolio/reduce-text',
        title: 'Reduce text',
        imgLight: '/images/reduce-light.png',
        imgDark: '/images/reduce-dark.png'
    },
    {
        url: '/portfolio/pokemon',
        title: 'Pokem√≥n API',
        imgLight: '/images/pokemon-light.png',
        imgDark: '/images/pokemon-dark.png'
    },
    {
        url: '/portfolio/fquiz',
        title: 'Fquiz App',
        imgLight: '/images/fquiz-light.png',
        imgDark: '/images/fquiz-dark.png'
    }
];
