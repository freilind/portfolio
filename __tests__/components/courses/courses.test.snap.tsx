import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { CourseList } from "../../../components/courses";

describe("Courses snapshot", () => {
  it("renders courses unchanged", () => {
    const { container } = render(<CourseList />);
    expect(container).toMatchSnapshot();
  });
});
