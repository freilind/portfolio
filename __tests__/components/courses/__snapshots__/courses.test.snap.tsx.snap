// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Courses snapshot renders courses unchanged 1`] = `
<div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-idYPjDv-css nextui-grid-item sm"
  >
    <a
      class="nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link"
      href="/images/courses/cert-java8.jpg"
      target="_blank"
    >
      <div
        class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ifVYFGe-css nextui-image-container nextui-image--loading"
        data-state="loading"
      >
        <div
          class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
        />
        <img
          alt="certificate-java8"
          class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
          data-state="loading"
          height="220"
          src="/images/courses/cert-java8.jpg"
          width="310"
        />
      </div>
    </a>
  </div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-idYPjDv-css nextui-grid-item sm"
  >
    <a
      class="nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link"
      href="https://udemy-certificate.s3.amazonaws.com/pdf/UC-625fca48-56d0-499a-abd0-2e3299aeb3a0.pdf"
      target="_blank"
    >
      <div
        class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ifVYFGe-css nextui-image-container nextui-image--loading"
        data-state="loading"
      >
        <div
          class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
        />
        <img
          alt="next-frmwk-prod"
          class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
          data-state="loading"
          height="220"
          src="/images/courses/next-frmwk-prod.jpeg"
          width="310"
        />
      </div>
    </a>
  </div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-idYPjDv-css nextui-grid-item sm"
  >
    <a
      class="nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link"
      href="https://udemy-certificate.s3.amazonaws.com/pdf/UC-ea89ede5-8ad5-4182-bc23-d467a82f7ab4.pdf"
      target="_blank"
    >
      <div
        class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ifVYFGe-css nextui-image-container nextui-image--loading"
        data-state="loading"
      >
        <div
          class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
        />
        <img
          alt="next-react-guide"
          class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
          data-state="loading"
          height="220"
          src="/images/courses/next-react-guide.jpeg"
          width="310"
        />
      </div>
    </a>
  </div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-idYPjDv-css nextui-grid-item sm"
  >
    <a
      class="nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link"
      href="https://udemy-certificate.s3.amazonaws.com/pdf/UC-fa37664f-a127-40f3-b590-64893ff75dee.pdf"
      target="_blank"
    >
      <div
        class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ifVYFGe-css nextui-image-container nextui-image--loading"
        data-state="loading"
      >
        <div
          class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
        />
        <img
          alt="web-bootcamp2022"
          class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
          data-state="loading"
          height="220"
          src="/images/courses/web-bootcamp2022.jpeg"
          width="310"
        />
      </div>
    </a>
  </div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-idYPjDv-css nextui-grid-item sm"
  >
    <a
      class="nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link"
      href="https://udemy-certificate.s3.amazonaws.com/pdf/UC-2486bc7f-f452-4c96-8f0d-9c9e229a7223.pdf"
      target="_blank"
    >
      <div
        class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ifVYFGe-css nextui-image-container nextui-image--loading"
        data-state="loading"
      >
        <div
          class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
        />
        <img
          alt="master-cqrs"
          class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
          data-state="loading"
          height="220"
          src="/images/courses/master-cqrs.jpeg"
          width="310"
        />
      </div>
    </a>
  </div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-idYPjDv-css nextui-grid-item sm"
  >
    <a
      class="nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link"
      href="https://udemy-certificate.s3.amazonaws.com/pdf/UC-5b81bda1-1845-4530-92df-681e05705f46.pdf"
      target="_blank"
    >
      <div
        class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ifVYFGe-css nextui-image-container nextui-image--loading"
        data-state="loading"
      >
        <div
          class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
        />
        <img
          alt="flutter-complete-guide"
          class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
          data-state="loading"
          height="220"
          src="/images/courses/flutter-complete-guide.jpeg"
          width="310"
        />
      </div>
    </a>
  </div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-idYPjDv-css nextui-grid-item sm"
  >
    <a
      class="nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link"
      href="https://udemy-certificate.s3.amazonaws.com/pdf/UC-07aabcbd-8c1e-461d-9518-e69cbf0d998c.pdf"
      target="_blank"
    >
      <div
        class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ifVYFGe-css nextui-image-container nextui-image--loading"
        data-state="loading"
      >
        <div
          class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
        />
        <img
          alt="flutter-2021"
          class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
          data-state="loading"
          height="220"
          src="/images/courses/flutter-2021.jpeg"
          width="310"
        />
      </div>
    </a>
  </div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-idYPjDv-css nextui-grid-item sm"
  >
    <a
      class="nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link"
      href="https://udemy-certificate.s3.amazonaws.com/pdf/UC-d4ef5915-4608-46ba-bfd4-f7baf95d2346.pdf"
      target="_blank"
    >
      <div
        class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ifVYFGe-css nextui-image-container nextui-image--loading"
        data-state="loading"
      >
        <div
          class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
        />
        <img
          alt="docker-guide"
          class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
          data-state="loading"
          height="220"
          src="/images/courses/docker-guide.jpeg"
          width="310"
        />
      </div>
    </a>
  </div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-idYPjDv-css nextui-grid-item sm"
  >
    <a
      class="nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link"
      href="https://udemy-certificate.s3.amazonaws.com/pdf/UC-bca7b665-586d-4d9c-8105-d0546e5a3dad.pdf"
      target="_blank"
    >
      <div
        class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ifVYFGe-css nextui-image-container nextui-image--loading"
        data-state="loading"
      >
        <div
          class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
        />
        <img
          alt="kafka"
          class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
          data-state="loading"
          height="220"
          src="/images/courses/kafka.jpeg"
          width="310"
        />
      </div>
    </a>
  </div>
</div>
`;
